source/frontend/src/features/properties/map/CompensationRequisitionRouter.tsx

```diff
@@ -0,0 +1,59 @@
+import Claims from 'constants/claims';
+import React from 'react';
+import { matchPath, Switch, useHistory, useLocation } from 'react-router';
+import AppRoute from 'utils/AppRoute';
+
+import { CompensationRequisitionTrayContainer } from './compensation/CompensationRequisitionTrayContainer';
+import { CompensationRequisitionTrayView } from './compensation/CompensationRequisitionTrayView';
+
+interface ICompensationRequisitionRouterProps {
+  setShowActionBar: (show: boolean) => void;
+}
+
+export const CompensationRequisitionRouter: React.FunctionComponent<
+  React.PropsWithChildren<ICompensationRequisitionRouterProps>
+> = React.memo(props => {
+  const location = useLocation();
+  const history = useHistory();
+
+  let matched = matchPath(location.pathname, {
+    path: '/mapview/sidebar/acquisition/*/compensation-requisition/*',
+    exact: true,
+    strict: true,
+  });
+
+  React.useEffect(() => {
+    if (matched !== null) {
+      props.setShowActionBar(true);
+    } else {
+      props.setShowActionBar(false);
+    }
+  }, [matched, props]);
+
+  const onClose = () => {
+    const backUrl = history.location.pathname.split('compensation-requisition')[0];
+    props.setShowActionBar(false);
```

Comment:
do we need both of these or will updating the url cause the useEffect to do this?

Reply:
updated to just one call.